enhancements/network/routed-ingress-support-for-primary-udn-attached-vms-with-static-ips.md

```diff
@@ -0,0 +1,789 @@
+feature.
+
+#### Standalone Clusters
+
+Full support.
+
+#### Single-node Deployments or MicroShift
+
+Full support; there might be some increased resource usage when scaling up
+number of VMs requesting IP and MAC address, since it will mean more processing
+in OVN-Kubernetes, and more CRs to reconcile.
+
+### Implementation Details/Notes/Constraints
+
+TODO
+
+### Risks and Mitigations
+
+TODO
+
+### Drawbacks
+
+The biggest drawback we have to implementing this feature is lack of clear
+asks from customers - we're pretty much guessing what they would want to use.
+
+We do not know (for instance):
+- is this feature only for VMs ?
+- is this feature only about importing VMs ? Should we allow creating new VMs
+  with dedicated MAC / IP / gateway requests ?
+- are the end users (admins) after a centralized IP management alternative ?
+  Do they want to have a single place to check their network address
+  assignments ?
+
+## Alternatives (Not Implemented)
+
+### Requesting IP addresses on a VM's interfaces
+An option which would provide a simpler implementation (and better / more
+explicit UX) would be to modify the VMI specification, adding an attribute for
+the user (or the tool templating the VM - e.g. MTV) to specify which IP
+addresses they want the VM's interface to take.
+
+It would be updating the
+[v1.Interface](https://kubevirt.io/api-reference/main/definitions.html#_v1_interface)
+with an ipAddresses array. This attribute is defined in
+`VMI.Spec.Domain.Devices.Interfaces`.
+
+This option (despite being the simplest to implement) is not taken into
+consideration since we're already past the KubeVirt 1.5 API freeze - as a
+result, we wouldn't be able to get this in OpenShift 4.20. This should be taken
+into account for a future release though.
+
+### Defining the MAC to IPs association
+There are two alternatives to persist this information: a
+[centralized approach](#centralized-ip-management),
+which has the benefits of a single centralized place an admin type of user
+could provision / get information from, or a
+[de-centralized approach](#de-centralized-ip-management), similar
+to the existing `IPAMClaim` CRD, where each VM connection to a primary UDN
+information would be recorded.
+
+Independently of which option we take, the API between OpenShift Virt (CNV) and
+OVN-Kubernetes will be the same, and is roughly described in
+[CNV to OVN-Kubernetes API](#cnv-ovnk-api).
+
+#### Centralized IP management
+
+A new CRD - named `IPPool`, or `DHCPLeaseConfig` (or the like) - will be
+created, and is associated to a cluster UDN. This CRD holds the association of
+MAC address to IPs for a UDN. When importing the VM into OpenShift Virt, MTV
+(or a separate, dedicated component) will provision / update this object with
+the required information (MAC to IPs association).
+This object is providing to the admin user a single place to check the IP
+address MAC to IPs mapping. On an first implementation phase, we can have the
```

Comment:
> check the IP address MAC to IPs mapping

Typo?
Maybe you mean "_check the MAC-to-IPs mapping_"?
(Or maybe it's me  misunderstanding what you mean here).